android/engine/src/main/java/org/smartregister/fhircore/engine/ui/pin/PinSetupScreen.kt

```diff
@@ -0,0 +1,211 @@
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.smartregister.fhircore.engine.R
+import org.smartregister.fhircore.engine.ui.components.PIN_INPUT_MAX_THRESHOLD
+import org.smartregister.fhircore.engine.ui.components.PinView
+import org.smartregister.fhircore.engine.ui.login.APP_LOGO_TAG
+import org.smartregister.fhircore.engine.util.annotation.ExcludeFromJacocoGeneratedReport
+
+@Composable
+fun PinSetupScreen(viewModel: PinViewModel) {
+
+  val inputPin by viewModel.pin.observeAsState(initial = "")
+  val enableSetPin by viewModel.enableSetPin.observeAsState(initial = false)
+
+  PinSetupPage(
+    onPinChanged = viewModel::onPinChanged,
+    inputPin = inputPin,
+    setPinEnabled = enableSetPin ?: false,
+    onPinConfirmed = viewModel::onPinConfirmed,
+    onMenuSettingClicked = { viewModel.onMenuSettingClicked() },
+    appName = viewModel.appName
+  )
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun PinSetupPage(
+  modifier: Modifier = Modifier,
+  onPinChanged: (String) -> Unit,
+  inputPin: String,
+  setPinEnabled: Boolean = false,
+  onPinConfirmed: () -> Unit,
+  onMenuSettingClicked: () -> Unit,
+  appName: String = ""
+) {
+
+  var showMenu by remember { mutableStateOf(false) }
+
+  Surface(color = colorResource(id = R.color.white_slightly_opaque)) {
+    TopAppBar(
+      title = { Text(text = "", Modifier.testTag(PIN_TOOLBAR_TITLE)) },
+      navigationIcon = {
+        IconButton(onClick = {}) {
+          Icon(
+            Icons.Filled.ArrowBack,
+            contentDescription = "Back arrow",
+            modifier = Modifier.size(0.dp).testTag(PIN_TOOLBAR_MENU_ICON)
+          )
+        }
+      },
+      actions = {
+        IconButton(
+          onClick = { showMenu = !showMenu },
+          modifier = Modifier.testTag(PIN_TOOLBAR_MENU_BUTTON)
+        ) { Icon(imageVector = Icons.Outlined.MoreVert, contentDescription = null) }
+        DropdownMenu(
+          expanded = showMenu,
+          onDismissRequest = { showMenu = false },
+          Modifier.testTag(PIN_TOOLBAR_MENU)
+        ) {
+          DropdownMenuItem(
+            onClick = {
+              showMenu = false
+              onMenuSettingClicked()
+            },
+            modifier = Modifier.testTag(PIN_TOOLBAR_MENU_SETTINGS)
+          ) { Text(text = stringResource(id = R.string.settings)) }
+        }
+      }
+    )
+
+    Column(
+      modifier =
+        Modifier.fillMaxSize()
+          .padding(horizontal = 16.dp, vertical = 70.dp)
+          .wrapContentWidth(Alignment.CenterHorizontally)
+    ) {
+      val appLogo =
+        when (appName) {
+          "g6pd" -> R.drawable.ic_logo_g6pd
```

Comment:
same comment as here, https://github.com/opensrp/fhircore/pull/1030/files#diff-3b37312122b63c666247d805d25fa24f597ae98f032a91bccea05285fc52aa08R156, also unrelated to the config refactor, this looks like it repeats the above code, can some of this be refactored to not duplicate?